bpf/mb_connect.bpf.c

```diff
@@ -28,8 +28,18 @@ struct {
     __uint(pinning, LIBBPF_PIN_BY_NAME);
 } cookie_orig_dst SEC(".maps");
 
+// BPF_MAP_TYPE_LRU_HASH - an LRU hash will
+// automatically evict the least recently used
+// entries when the hash table reaches capacity
+// We set the size of 65535 to cover number of
+// pods that can run on one node. We want to cover
+// networks with mask x.x.x.x/16 which allows to
+// run 65535 unique pods, and if new one appears
+// the oldest not accessed entry is going to be
+// removed from the map by the kernel. This ensure
+// that configuration of living pod is removed.
```

Comment:
Random question could there be a problem with a pod receiving 0 traffic?

Reply:
Shouldn't be, if pod doesn't receive traffic the only situation in which it might be removed is when we deploy > 65535 pods in the cluster

Reply:
Yes but we're not talking about simultaneous pods right? So for example in a cluster that has batch jobs we create new pods all the time (that are short lived). Could this potentially be a problem?

This doesn't have to be fixed now as it's an edge case but might be worth at least understanding if it's possible

Reply:
Yes, that might be the edge case. 65k is the number of pods per node - not cluster. But true, if we have a lot of short-living pods their configuration is placed in the map and if we have a pod without any traffic(even healthcheck) it might cause this.